vllm/attention/backends/hpu_attn.py

```diff
@@ -261,6 +284,118 @@ def forward(
         # Reshape the output tensor.
         return output.view(batch_size, seq_len, hidden_size)
 
+    def forward_encoder_decoder(
+        self,
+        query: torch.Tensor,
+        key: torch.Tensor,
+        value: torch.Tensor,
+        kv_cache: torch.Tensor,
+        attn_metadata: HPUAttentionMetadata,
+        k_scale: float = 1.0,
+        v_scale: float = 1.0,
+    ) -> torch.Tensor:
+        """Forward pass with xFormers and PagedAttention.
+
+        Args:
+            query: shape = [num_tokens, num_heads * head_size]
+            key: shape = [num_tokens, num_kv_heads * head_size]
+            value: shape = [num_tokens, num_kv_heads * head_size]
+            kv_cache = [2, num_blocks, block_size * num_kv_heads * head_size]
+            attn_metadata: Metadata for attention.
+        Returns:
+            shape = [num_tokens, num_heads * head_size]
+        """
+        batch_size, _ = query.shape
+
+        if attn_metadata.is_prompt:
+            batch_size = attn_metadata.num_prefills
+            batched_tokens, _ = query.shape
+            batched_kv_tokens, _, _ = key.shape
+            assert batch_size > 0, (
+                "In prefill stage the num_prefills should be > 0")
+            assert batched_tokens % batch_size == 0
+            assert batched_kv_tokens % batch_size == 0
```

Comment:
Are any of those asserts is even possible? It looks like redundant checks to me

Reply:
Yes, in different image sizes input scenario, which will be implemented in the next pr.